test/com/google/javascript/jscomp/AstValidatorTest.java

```diff
@@ -1265,6 +1265,38 @@ public void testUnresolvedTypesAreBanned() {
     expectValid(expr, Check.STATEMENT);
   }
 
+  @Test
+  public void testSwitchStatement() {
+    // Since we're building the AST by hand, there won't be any types on it.
+    enableTypeInfoValidation = false;
+
+    String switchStatement = lines(
+        "function foo(x) {",
+        "  switch(x) {",
+        "    case 1: ",
+        "      var y = 5;",
+        "    default:",
```

Comment:
Is `default` here to to make this expression invalid? Or you add `defaultBlock` node later to make the whole tree invalid? Not sure what is the point of `defaultBlock` below.

Reply:
No, there is no reason for this. I deleted it to make it more clear. I also changed the `defaultBlock` to just `blockNode`. This node is added to give an extra child where it should not be allowed and that makes it invalid.